lib/stats/DefaultStatsFactoryPlugin.js

```diff
@@ -1779,6 +1781,46 @@ const spaceLimited = (
 	};
 };
 
+const errorsSpaceLimit = (errors, max) => {
+	if (!Number.isFinite(max)) return errors;
+	// Can not fit into limit
+	// print only messages
+	if (errors.length > max)
+		return errors.map(error => {
+			if (error.details === undefined) return error;
+			const { details, ...rest } = error;
+			return { details: "", ...rest };
+		});
+
+	let fullLength = errors.length;
+	let result = errors;
+
+	let i = 0;
+	for (; i < errors.length; i++) {
+		if (errors[i].details) {
+			const splitted = errors[i].details.split("\n");
+			const len = splitted.length;
+			fullLength += len;
+			if (fullLength > max) {
+				result = errors.slice(0, i);
+				const { details, ...rest } = errors[i++];
+				result.push({
+					details: splitted.slice(0, len - (fullLength - max)).join("\n"),
+					...rest
+				});
```

Comment:
```suggestion
				result.push({
				  ...errors[i++],
					details: splitted.slice(0, len - (fullLength - max)).join("\n")
				});
```

No need to create a temporary `rest` object.